sds/src/scanner/mod.rs

```diff
@@ -514,6 +523,15 @@ impl Scanner {
         self.metrics
             .match_count
             .increment(output_rule_matches.len() as u64);
+
+        if let Some(io_duration) = io_duration {
+            let total_duration = start.elapsed();
+            let cpu_duration = total_duration.saturating_sub(io_duration);
+            // Record CPU duration counter in nanoseconds
```

Comment:
self explanatory code, if you want to be more explicit, you can rename the variable `cpu_duration_ns` and the metric name `scanning.cpu_duration_ns`
```suggestion
```